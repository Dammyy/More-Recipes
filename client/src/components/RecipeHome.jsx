import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';

const BtnView = ((props) => {
  return (
    <Link
      className="btn btn-success btn-view"
      to={`view/${props.id}`}
      params={{ id: props.id }}
    >
      View
    </Link >);
});
/**
 *
 *
 * @export
 * @class RecipeHome
 * @extends {PureComponent}
 */
class RecipeHome extends PureComponent {
  /**
   *
   *
   * @returns {void}
   *
   * @memberOf RecipeHome
   */
  render() {
    const {
      id, title, image, upvotes, downvotes, favorited
    } = this.props;
    let img;
    if (image === '') {
      img = 'http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg';
    } else {
      img = image;
    }
    return (
      <div className="col-md-4 recipe-display">
        <img src={img} alt="" />
        <div id="votes-del">
          <div id="recipe-votes">
            <li>
              <i className="fa fa-heart" /> {favorited}
            </li>
            <li>
              <i className="fa fa-thumbs-o-up" aria-hidden="true"> {upvotes}
              </i>
            </li>
            <li>
              <i className="fa fa-thumbs-o-down" aria-hidden="true"> {downvotes}
              </i>
            </li>
          </div>
        </div>
        <div id="recipe-title"><h2>{title}</h2></div>
        <BtnView id={id} />
      </div>
    );
  }
}
RecipeHome.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  upvotes: PropTypes.number.isRequired,
  downvotes: PropTypes.number.isRequired,
  favorited: PropTypes.number.isRequired
};

BtnView.propTypes = {
  id: PropTypes.number.isRequired,
};
export default RecipeHome;
