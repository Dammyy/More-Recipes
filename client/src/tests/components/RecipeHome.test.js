import React from 'react';
import { shallow } from 'enzyme';
import RecipeHome, { BtnView } from '../../components/RecipeHome';

describe('BtnView Snapshot', () => {
  test('component should fully render', () => {
    const props = {
      id: 1
    };
    const component = shallow(<BtnView {...props} />);
    expect(component).toMatchSnapshot();
  });
});

describe('RecipeHomeSnapshot', () => {
  test('component should fully render(image set)', () => {
    const props = {
      id: 1,
      title: '',
      image: 'http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg',
      favorited: 1,
      upvotes: 1,
      downvotes: 1
    };
    const component = shallow(<RecipeHome{...props} />);
    expect(component).toMatchSnapshot();
  });
});

describe('RecipeHomeSnapshot', () => {
  test('component should fully render(image not set)', () => {
    const props = {
      id: 1,
      title: '',
      image: '',
      favorited: 1,
      upvotes: 1,
      downvotes: 1
    };
    const component = shallow(<RecipeHome{...props} />);
    expect(component).toMatchSnapshot();
  });
});
