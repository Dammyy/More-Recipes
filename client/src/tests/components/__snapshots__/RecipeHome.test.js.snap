// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnView Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BtnView
    id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "View",
      "className": "btn btn-success btn-view",
      "onlyActiveOnIndex": false,
      "params": Object {
        "id": 1,
      },
      "style": Object {},
      "to": "view/1",
    },
    "ref": null,
    "rendered": "View",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "View",
        "className": "btn btn-success btn-view",
        "onlyActiveOnIndex": false,
        "params": Object {
          "id": 1,
        },
        "style": Object {},
        "to": "view/1",
      },
      "ref": null,
      "rendered": "View",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`RecipeHomeSnapshot component should fully render(image not set) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeHome
    downvotes={1}
    favorited={1}
    id={1}
    image=""
    title=""
    upvotes={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg"
        />,
        <div
          id="votes-del"
        >
          <div
            id="recipe-votes"
          >
            <li>
              <i
                className="fa fa-heart"
              />
               
              1
            </li>
            <li>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-o-up"
              >
                 
                1
              </i>
            </li>
            <li>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-o-down"
              >
                 
                1
              </i>
            </li>
          </div>
        </div>,
        <div
          id="recipe-title"
        >
          <h2>
            
          </h2>
        </div>,
        <BtnView
          id={1}
        />,
      ],
      "className": "col-md-4 recipe-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            id="recipe-votes"
          >
            <li>
              <i
                className="fa fa-heart"
              />
               
              1
            </li>
            <li>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-o-up"
              >
                 
                1
              </i>
            </li>
            <li>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-o-down"
              >
                 
                1
              </i>
            </li>
          </div>,
          "id": "votes-del",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <i
                  className="fa fa-heart"
                />
                 
                1
              </li>,
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                  1
                </i>
              </li>,
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                  1
                </i>
              </li>,
            ],
            "id": "recipe-votes",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fa fa-heart"
                  />,
                  " ",
                  1,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-heart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " ",
                1,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                  1
                </i>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "children": Array [
                    " ",
                    1,
                  ],
                  "className": "fa fa-thumbs-o-up",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  1,
                ],
                "type": "i",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                  1
                </i>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "children": Array [
                    " ",
                    1,
                  ],
                  "className": "fa fa-thumbs-o-down",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  1,
                ],
                "type": "i",
              },
              "type": "li",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            
          </h2>,
          "id": "recipe-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg"
          />,
          <div
            id="votes-del"
          >
            <div
              id="recipe-votes"
            >
              <li>
                <i
                  className="fa fa-heart"
                />
                 
                1
              </li>
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                  1
                </i>
              </li>
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                  1
                </i>
              </li>
            </div>
          </div>,
          <div
            id="recipe-title"
          >
            <h2>
              
            </h2>
          </div>,
          <BtnView
            id={1}
          />,
        ],
        "className": "col-md-4 recipe-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              id="recipe-votes"
            >
              <li>
                <i
                  className="fa fa-heart"
                />
                 
                1
              </li>
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                  1
                </i>
              </li>
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                  1
                </i>
              </li>
            </div>,
            "id": "votes-del",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <i
                    className="fa fa-heart"
                  />
                   
                  1
                </li>,
                <li>
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-o-up"
                  >
                     
                    1
                  </i>
                </li>,
                <li>
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-o-down"
                  >
                     
                    1
                  </i>
                </li>,
              ],
              "id": "recipe-votes",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-heart"
                    />,
                    " ",
                    1,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-heart",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                  1,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-thumbs-o-up"
                  >
                     
                    1
                  </i>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": Array [
                      " ",
                      1,
                    ],
                    "className": "fa fa-thumbs-o-up",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    1,
                  ],
                  "type": "i",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-thumbs-o-down"
                  >
                     
                    1
                  </i>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": Array [
                      " ",
                      1,
                    ],
                    "className": "fa fa-thumbs-o-down",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    1,
                  ],
                  "type": "i",
                },
                "type": "li",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              
            </h2>,
            "id": "recipe-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`RecipeHomeSnapshot component should fully render(image set) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeHome
    downvotes={1}
    favorited={1}
    id={1}
    image="http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg"
    title=""
    upvotes={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg"
        />,
        <div
          id="votes-del"
        >
          <div
            id="recipe-votes"
          >
            <li>
              <i
                className="fa fa-heart"
              />
               
              1
            </li>
            <li>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-o-up"
              >
                 
                1
              </i>
            </li>
            <li>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-o-down"
              >
                 
                1
              </i>
            </li>
          </div>
        </div>,
        <div
          id="recipe-title"
        >
          <h2>
            
          </h2>
        </div>,
        <BtnView
          id={1}
        />,
      ],
      "className": "col-md-4 recipe-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            id="recipe-votes"
          >
            <li>
              <i
                className="fa fa-heart"
              />
               
              1
            </li>
            <li>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-o-up"
              >
                 
                1
              </i>
            </li>
            <li>
              <i
                aria-hidden="true"
                className="fa fa-thumbs-o-down"
              >
                 
                1
              </i>
            </li>
          </div>,
          "id": "votes-del",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <i
                  className="fa fa-heart"
                />
                 
                1
              </li>,
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                  1
                </i>
              </li>,
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                  1
                </i>
              </li>,
            ],
            "id": "recipe-votes",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fa fa-heart"
                  />,
                  " ",
                  1,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-heart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " ",
                1,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                  1
                </i>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "children": Array [
                    " ",
                    1,
                  ],
                  "className": "fa fa-thumbs-o-up",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  1,
                ],
                "type": "i",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                  1
                </i>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "children": Array [
                    " ",
                    1,
                  ],
                  "className": "fa fa-thumbs-o-down",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  1,
                ],
                "type": "i",
              },
              "type": "li",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            
          </h2>,
          "id": "recipe-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg"
          />,
          <div
            id="votes-del"
          >
            <div
              id="recipe-votes"
            >
              <li>
                <i
                  className="fa fa-heart"
                />
                 
                1
              </li>
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                  1
                </i>
              </li>
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                  1
                </i>
              </li>
            </div>
          </div>,
          <div
            id="recipe-title"
          >
            <h2>
              
            </h2>
          </div>,
          <BtnView
            id={1}
          />,
        ],
        "className": "col-md-4 recipe-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "http://wyregate.com/wp-content/uploads/2018/03/recipe.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              id="recipe-votes"
            >
              <li>
                <i
                  className="fa fa-heart"
                />
                 
                1
              </li>
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                  1
                </i>
              </li>
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                  1
                </i>
              </li>
            </div>,
            "id": "votes-del",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <i
                    className="fa fa-heart"
                  />
                   
                  1
                </li>,
                <li>
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-o-up"
                  >
                     
                    1
                  </i>
                </li>,
                <li>
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-o-down"
                  >
                     
                    1
                  </i>
                </li>,
              ],
              "id": "recipe-votes",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-heart"
                    />,
                    " ",
                    1,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-heart",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                  1,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-thumbs-o-up"
                  >
                     
                    1
                  </i>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": Array [
                      " ",
                      1,
                    ],
                    "className": "fa fa-thumbs-o-up",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    1,
                  ],
                  "type": "i",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-thumbs-o-down"
                  >
                     
                    1
                  </i>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": Array [
                      " ",
                      1,
                    ],
                    "className": "fa fa-thumbs-o-down",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    1,
                  ],
                  "type": "i",
                },
                "type": "li",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              
            </h2>,
            "id": "recipe-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
